### ISSUE TITLE
Warning or Info why schedule stopped before reaching the final step.

### ISSUE THREAD

**Issue by JohannesK14** [speediedan/finetuning-scheduler#7]
## 🚀 Feature

I would like to see finetuning-scheduler logging or warning (or at least an info message) if it ends because the parameter max_depth was set to something smaller than the number of available finetuning steps.

### Motivation

I accidentally set the max_depth parameter to 1 in one of my config files. Consequently, the training finished after stage 1 of my schedule. But it did so without some info/ warning. It took me quite some time to track down the wrong config setting. 

### Pitch

Send an info or warning message if the training is stopped because of max_depth being smaller than the number of available steps.

### Alternatives

Leave it as it is ...

### Additional context

______________________________________________________________________

**Comment by JohannesK14** [comment]
Thanks!

**Comment by speediedan** [comment]
Great suggestion! Thanks for your valuable feedback and making FTS better. I'm delighted you're finding FTS of at least some utility @JohannesK14 

By the way, FTS `0.4.0` has just been released with initial support for **Multi-phase Scheduled Fine-Tuning with FSDP** 🎉 [See the FSDP tutorial here if interested](https://finetuning-scheduler.readthedocs.io/en/stable/advanced/fsdp_scheduled_fine_tuning.html). Also, support for PyTorch 2.0 is planned as soon as it is stable so should be there with the next non-patch release of FTS.
